tests/FSharp.Compiler.ComponentTests/Miscellaneous/FsharpSuiteMigrated.fs

```diff
@@ -0,0 +1,174 @@
+let exit (code:int) = 
+    System.Console.SetError(oldConsoleError)
+    if code=0 then 
+        () 
+    else failwith $"Script called function 'exit' with code={code} and collected in stderr: {errorStringWriter.ToString()}" """ |> ignore
+        scriptingEnv
+
+    let defaultDefines = 
+        [ 
+#if NETCOREAPP
+          "NETCOREAPP"
+#endif
+        ]
+
+    let runScriptFile version (cu:CompilationUnit)  =
+        let cu  = cu |> withDefines defaultDefines
+        match cu with 
+        | FS fsSource ->
+            File.Delete("test.ok")           
+            let engine = createEngine (fsSource.Options |> Array.ofList,version)
+            let res = evalScriptFromDiskInSharedSession engine cu
+            match res with
+            | CompilationResult.Failure _ -> res
+            | CompilationResult.Success s -> 
+                if File.Exists("test.ok") then
+                    res
+                else
+                    failwith $"Results looked correct, but 'test.ok' file was not created. Result: %A{s}"       
+
+        | _ -> failwith $"Compilation unit other than fsharp is not supported, cannot process %A{cu}"
+
+/// This test file was created by porting over (slower) FsharpSuite.Tests
+/// In order to minimize human error, the test definitions have been copy-pasted and this adapter provides implementations of the test functions
+module TestFrameworkAdapter = 
+    open FSharp.Test.Compiler.Assertions.TextBasedDiagnosticAsserts
+
+    type ExecutionMode = 
+        | FSC_DEBUG 
+        | FSC_OPTIMIZED 
+        | FSI
+        | COMPILED_EXE_APP
+        | NEG_TEST_BUILD of testName:string
+
+    let baseFolder = Path.Combine(__SOURCE_DIRECTORY__,"..","..","fsharp") |> Path.GetFullPath      
+
+    let diffNegativeBaseline (cr:CompilationUnit) absFolder testName _version  =
+        let expectedFiles = Directory.GetFiles(absFolder, testName + ".*")
+        let baselines = 
+            [ for f in expectedFiles do
+                match Path.GetExtension(f) with
+                | ".bsl" -> cr, f
+                | ".vsbsl" -> cr |> withOptions ["--test:ContinueAfterParseFailure"], f
+                | _ -> () ]
+        [ for compilationUnit,baseline in baselines do
+            compilationUnit       
+            |> typecheck
+            |> withResultsMatchingFile baseline ]
+        |> List.head
+            
+
+    let adjustVersion version bonusArgs = 
+        match version with 
+        | LangVersion.V47 -> "4.7",bonusArgs
+        | LangVersion.V50 -> "5.0",bonusArgs
+        | LangVersion.V60 -> "6.0",bonusArgs
+        | LangVersion.V70 -> "7.0",bonusArgs
+        | LangVersion.Preview -> "preview",bonusArgs
+        | LangVersion.Latest  -> "latest", bonusArgs
+        | LangVersion.SupportsMl -> "5.0",  "--mlcompatibility" :: bonusArgs       
+
+    let supportedNames = set ["testlib.fsi";"testlib.fs";"test.mli";"test.ml";"test.fsi";"test.fs";"test2.fsi";"test2.fs";"test.fsx";"test2.fsx"]
```

Comment:
This looks extremely subtle, I'd imagine at least hardcoding this on some more outer level...

Reply:
What does outer level mean?
This is a top-level module shared across migrated tests, cannot get much more outer than that.

Reply:
I'd probably just create an extra module for values like this - and reference them from elsewhere.